packages/components/src/internal/components/picklist/PicklistEditModal.tsx

```diff
@@ -151,14 +152,15 @@ export const PicklistEditModalDisplay: FC<Props> = memo(props => {
         });
     };
 
-    const onSavePicklist = useCallback(async () => {
+    const onSavePicklist = async (): Promise<void> => {
```

Comment:
Is there a particular reason this is no longer a useCallback?

Reply:
It was due to complexity. The previously stated dependencies array for this hook were not correct and only declared a small subset of the actual dependencies. Here is what this would need to look like in it's current form as a `useCallback`:

```ts
const onSavePicklist = useCallback(async (): Promise<void> => {
    setIsSubmitting(true);
    try {
        let updatedList;
        const trimmedName = name.trim();
        if (isUpdate) {
            updatedList = await updatePicklist(
                new Picklist({
                    Container: picklist.Container,
                    name: trimmedName,
                    listId: picklist.listId,
                    Description: description,
                    Category: shared ? PUBLIC_PICKLIST_CATEGORY : PRIVATE_PICKLIST_CATEGORY,
                })
            );
        } else {
            updatedList = await createPicklist(
                trimmedName,
                description,
                shared,
                statusData,
                selectionKey,
                sampleIds
            );
            api.query.incrementClientSideMetricCount(metricFeatureArea, 'createPicklist');
        }
        reset();
        if (showNotification) {
            createSuccessNotification(updatedList);
        }

        onFinish(updatedList);
    } catch (e) {
        setPicklistError(resolveErrorMessage(e));
        setIsSubmitting(false);
    }
}, [
    api.query,
    createSuccessNotification,
    description,
    isUpdate,
    metricFeatureArea,
    name,
    onFinish,
    picklist,
    sampleIds,
    selectionKey,
    shared,
    showNotification,
    statusData,
]);
```
The dependency set is sufficiently large enough that I don't think it warrants a check each time for performance savings.